cluster-autoscaler/utils/drain/drain_test.go

```diff
@@ -624,7 +642,73 @@ func TestDrain(t *testing.T) {
 		},
 	}
 
-	for _, test := range tests {
+	// run all tests for scaleDownNodesWithCustomControllerPods=false
+	// TODO(vadasambar): remove this when we get rid of scaleDownNodesWithCustomControllerPods
+	for _, test := range append(tests, testOpts{
+		// Expect blocking
+		description:         "Custom-controller-managed pod",
+		pods:                []*apiv1.Pod{customControllerPod},
+		pdbs:                []*policyv1.PodDisruptionBudget{},
+		expectFatal:         true,
+		expectPods:          []*apiv1.Pod{},
+		expectBlockingPod:   &BlockingPod{Pod: customControllerPod, Reason: NotReplicated},
+		expectDaemonSetPods: []*apiv1.Pod{},
+	}) {
+		var err error
+		var rcLister v1lister.ReplicationControllerLister
+		if len(test.rcs) > 0 {
+			rcLister, err = kube_util.NewTestReplicationControllerLister(test.rcs)
+			assert.NoError(t, err)
+		}
+		var rsLister v1appslister.ReplicaSetLister
+		if len(test.replicaSets) > 0 {
+			rsLister, err = kube_util.NewTestReplicaSetLister(test.replicaSets)
+			assert.NoError(t, err)
+		}
+
+		dsLister, err := kube_util.NewTestDaemonSetLister([]*appsv1.DaemonSet{&ds})
+		assert.NoError(t, err)
+		jobLister, err := kube_util.NewTestJobLister([]*batchv1.Job{&job})
+		assert.NoError(t, err)
+		ssLister, err := kube_util.NewTestStatefulSetLister([]*appsv1.StatefulSet{&statefulset})
+		assert.NoError(t, err)
+
+		registry := kube_util.NewListerRegistry(nil, nil, nil, nil, nil, dsLister, rcLister, jobLister, rsLister, ssLister)
+
+		pods, daemonSetPods, blockingPod, err := GetPodsForDeletionOnNodeDrain(test.pods, test.pdbs, true, true, false, registry, 0, testTime)
+
+		if test.expectFatal {
+			assert.Equal(t, test.expectBlockingPod, blockingPod)
+			if err == nil {
+				t.Fatalf("%s: unexpected non-error", test.description)
+			}
+		}
+
+		if !test.expectFatal {
+			assert.Nil(t, blockingPod)
+			if err != nil {
+				t.Fatalf("%s: error occurred: %v", test.description, err)
+			}
+		}
+
+		if len(pods) != len(test.expectPods) {
+			t.Fatalf("Wrong pod list content: %v", test.description)
+		}
+
+		assert.ElementsMatch(t, test.expectDaemonSetPods, daemonSetPods)
+	}
+
+	// run all tests for scaleDownNodesWithCustomControllerPods=true
+	for _, test := range append(tests, testOpts{
```

Comment:
Note that I am running the same tests twice. Code [here](https://github.com/kubernetes/autoscaler/pull/5507/files#diff-cd1a98b8de97414dad2b66a3cf3229b1572a169086d9b7141a266f8ddb597f68R656-R699) is practically the same as code beyond line 702 above except. The only difference is `scaleDownNodesWithCustomControllerPods` is set to false [here](https://github.com/kubernetes/autoscaler/pull/5507/files#diff-cd1a98b8de97414dad2b66a3cf3229b1572a169086d9b7141a266f8ddb597f68R678) and it is set to `true` in the test cases beyond the above line. 

I did think of abstracting duplicate code into a separate function but I am not so sure after thinking about the comment [here](https://github.com/kubernetes/autoscaler/pull/5507#discussion_r1135330465) since abstracting the tests into a function would add another level of indirection. I am a little on the fence about doing that now. Any suggestions here are welcome.